src/runtime_src/core/include/xrt/xrt_bo.h

```diff
@@ -111,18 +111,22 @@ class bo
    *  Create a BO for peer-to-peer use
    * @var svm
    *  Create a BO for SVM (supported on specific platforms only)
-   *
+   * @var carveout
+   *  Create a BO from a special memory pool dedicated to the device.
+   *  For AMD Ryzen NPU this memory is allocated from the host memory carveout pool.
+   * 
    * The flags used by xrt::bo are compatible with XCL style
    * flags as define in ``xrt_mem.h``
    */
   enum class flags : uint32_t
   {
-    normal      = 0,
-    cacheable   = XRT_BO_FLAGS_CACHEABLE,
-    device_only = XRT_BO_FLAGS_DEV_ONLY,
-    host_only   = XRT_BO_FLAGS_HOST_ONLY,
-    p2p         = XRT_BO_FLAGS_P2P,
-    svm         = XRT_BO_FLAGS_SVM,
+    normal       = 0,
+    cacheable    = XRT_BO_FLAGS_CACHEABLE,
+    device_only  = XRT_BO_FLAGS_DEV_ONLY,
+    host_only    = XRT_BO_FLAGS_HOST_ONLY,
+    p2p          = XRT_BO_FLAGS_P2P,
+    svm          = XRT_BO_FLAGS_SVM,
+    carveout     = XRT_BO_FLAGS_CARVEOUT,
```

Comment:
No, carveout is horrible as an API facing symbolic name.   I will discuss with Sonal.

Reply:
I did a bit of search, and carveout is used by android to indicated special heap memory pool created a boot time.  I backtrack and say we can use `carveout` after all.